backend/static/js/codemirror/modes/sparql/sparql-hint.js

```diff
@@ -797,6 +798,19 @@ function getQleverSuggestions(
     // Issue AC query (or two when in mixed mode) and get `response`.
     try {
       activeWebSockets.forEach(ws => ws.close());
+      const unregisterWebSocket = (ws) => {
+        // If still connecting, close once open.
+        if (ws.readyState === WebSocket.CONNECTING) {
+          const oldOnOpen = ws.onopen;
+          ws.onopen = () => {
+            oldOnOpen();
+            ws.close();
+          };
+        } else {
+          ws.close();
+        }
+        activeWebSockets.delete(ws);
```

Comment:
Can you briefly explain why that is necessary? And is this the simplest / most natural way to achieve that?